rust/src/client/application/analyzer.rs

```diff
@@ -0,0 +1,104 @@
+// Software Name: its-client
+// SPDX-FileCopyrightText: Copyright (c) 2016-2023 Orange
+// SPDX-License-Identifier: MIT License
+//
+// This software is distributed under the MIT license, see LICENSE.txt file for more details.
+//
+// Author: Nicolas BUFFON <nicolas.buffon@orange.com> et al.
+// Software description: This Intelligent Transportation Systems (ITS) [MQTT](https://mqtt.org/) client based on the [JSon](https://www.json.org) [ETSI](https://www.etsi.org/committee/its) specification transcription provides a ready to connect project for the mobility (connected and autonomous vehicles, road side units, vulnerable road users,...).
+
+use crate::client::configuration::Configuration;
+use crate::exchange::Exchange;
+use crate::transport::mqtt::topic::Topic;
+use crate::transport::packet::Packet;
+
+use crate::exchange::sequence_number::SequenceNumber;
+use std::sync::{Arc, RwLock};
+
+/// Structures implementing this trait can be used in [crate::client::application::pipeline::run]
+/// to treat messages and eventually send or update other ones
+///
+/// Analyzer implementing structs must be able to run in parallel to be able to treat messages
+/// faster than they arrive
```

Comment:
Perhaps a typing error here.
Should it be:
faster when they arrive
instead of
faster than they arrive